src/routines/write_results.jl

```diff
@@ -40,184 +48,239 @@ function _write_data(data::DataFrames.DataFrame, save_path::AbstractString, file
+    kwargs...,
+)
     if !isdir(folder_path)
-        throw(ArgumentError("Specified path is not valid. Run write_results to save results."))
+        throw(IS.ConflictingInputsError("Specified path is not valid. Run write_results to save results."))
     end
     _write_data(res.variables, res.time_stamp, folder_path; kwargs...)
-    _write_data(res.duals, folder_path; kwargs...)
+    _write_data(res.constraints_duals, folder_path; kwargs...)
     _write_optimizer_log(res.optimizer_log, folder_path)
     _write_data(res.time_stamp, folder_path, "time_stamp"; kwargs...)
-    _write_results_sum(res, folder_path)
+    files = collect(readdir(folder_path))
+    compute_file_hash(folder_path, files)
     @info("Files written to $folder_path folder.")
     return
 end
 
-""" 
-    write_model_results(results::OperationsProblemResults, save_path::String, results_folder::String)
+"""
+    write_results(results::SimulationResults, save_path::String, results_folder::String)
 
 Exports Simulations Results to the path where they come from in the results folder
 
 # Arguments
-- `results::Results`: results from the simulation
+- `results::SimulationResults`: results from the simulation
 - `save_path::String`: folder path where the files will be written
 - `results_folder`: name of the folder where the results will be written
 
 # Accepted Key Words
 - `file_type = CSV`: only CSV and featherfile are accepted
 """
 
-function write_results(res::CheckResults, folder_path::String, results_folder::String; kwargs...)
+function write_results(
+    res::SimulationResults,
+    folder_path::String,
+    results_folder::String;
+    kwargs...,
+)
     if !isdir(folder_path)
-        throw(ArgumentError("Specified path is not valid. Run write_results to save results."))
+        throw(IS.ConflictingInputsError("Specified path is not valid. Run write_results to save results."))
     end
     _write_data(res.variables, res.time_stamp, folder_path; kwargs...)
     _write_optimizer_log(res.optimizer_log, folder_path)
     _write_data(res.time_stamp, folder_path, "time_stamp"; kwargs...)
-    _write_results_sum(res, folder_path)
+    files = collect(readdir(folder_path))
+    compute_file_hash(folder_path, files)
     @info("Files written to $folder_path folder.")
     return
 end
 
+function write_results(
+    res::OperationsProblemResults,
+    folder_path::String,
+    results_folder::String;
+    kwargs...,
+)
+    if !isdir(folder_path)
+        throw(IS.ConflictingInputsError("Specified path is not valid. Run write_results to save results."))
+    end
+    _write_data(res.variables, res.time_stamp, folder_path; kwargs...)
+    _write_optimizer_log(res.optimizer_log, folder_path)
+    _write_data(res.time_stamp, folder_path, "time_stamp"; kwargs...)
+    files = collect(readdir(folder_path))
+    compute_file_hash(folder_path, files)
+    @info("Files written to $folder_path folder.")
+    return
+end
+
+function compute_file_hash(path::String, files::Array)
```

Comment:
```julia
function compute_file_hash(path::String, files::Vector{String})
```

Reply:
Also, the `compute_file_hash` function can be moved to `src/utils/utils.jl`.

Reply:
File re-organization needs to be done later.